src/extension/importer/rest/src/main/java/org/geoserver/importer/rest/ImportTaskController.java

```diff
@@ -208,7 +208,11 @@ public void targetPut(
             throw new RestException("Task has no target store", HttpStatus.NOT_FOUND);
         } else {
             updateStoreInfo(task(id, taskId), store, importer);
-            importer.changed(task(id, taskId));
+            try {
+                importer.changed(task(id, taskId));
+            } catch (IOException e) {
+                throw new RestException("Error while initializing Importer Context", HttpStatus.INTERNAL_SERVER_ERROR, e);
+            }
```

Comment:
For the same reasons above, I thought it could be better to stop the games suddenly if a lock went wrong. This why otherwise at the next cleanup cycle, with no valid lock files, the scheduler will blindly try to wipe out everything.